TAScheduler/UnitTests/tests.py

```diff
@@ -46,38 +52,476 @@ class TestAdminCreateSection(TestCase):  # Alec
 
 
 class TestAdminRemoveCourse(TestCase):  # Kevin
-    pass
+    tempCourse = None
+    admin = None
+    hold_course = None
+
+    def setUp(self):
+        self.hold_course = Course.objects.create(
+            course_id=101,
+            semester='Fall 2023',
+            name='Introduction to Testing',
+            description='A course about writing tests in Django.',
+            num_of_sections=3,
+            modality='Online',
+            credits=4
+        )
+        self.tempCourse = CourseObj(self.hold_course)
+        hold_user = User(
+            email_address='admin@example.com',
+            password='adminpassword',
+            first_name='Admin',
+            last_name='User',
+            home_address='123 Admin St',
+            phone_number='1234567890'
+        )
+        hold_user.save()
+        hold_admin = Administrator(user=hold_user)
+        hold_admin.save()
+        self.admin = AdminObj(hold_admin)
+
+    def test_successful_delete(self):
+        self.admin.removeCourse(self.tempCourse)
+        self.assertNotIn(Course.objects, self.hold_course, "Did not remove course from the database")
+
+    def test_delete_null_course(self):
+        Course.delete(self.hold_course)
+        with self.assertRaises(RuntimeError, msg="Tried to delete a non-existent course"):
+            self.admin.removeCourse(self.tempCourse)
+
+    def test_delete_non_course(self):
+        with self.assertRaises(TypeError, msg="Tried to delete not a course"):
+            self.admin.removeCourse(11)
```

Comment:
You could call Course.getAsgmtsForCrse() and just assert empty/failure. (Just an idea)

Reply:
I dont think that is needed because of how the database is structured. Once a course is gone all items with a foreign key to course deleted themself. Also I don't want to use methods from views_methods.py because that introduces more potential errors into the test itself.

Reply:
I still think you need to test for it somehow - use the junction tables in the db to avoid the viewsMethod.

Even though we "think" it's not needed (I agree) this is an important feature that should probably be confirmed with a UnitTest